src/ProductConstructionService/ProductConstructionService.Api/Controllers/InternalController.cs

```diff
@@ -0,0 +1,34 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Reflection;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Mvc.Controllers;
+
+namespace ProductConstructionService.Api.Controllers;
+
+/// <summary>
+/// Enables us to create internal controllers so that they can have internal dependencies.
+/// All controllers that inject internal dependencies should inherit from this class.
+/// </summary>
+public abstract class InternalController : Controller
+{
+}
+
+public static class InternalControllersExtension
```

Comment:
Should we really be doing this? Why do we want to make all controllers and their dependencies internal?

Reply:
So ideally, we don't want anything to be public that is used only inside of the API - like the job status or the background workers etc.
Once you make one of those internal, it triggers a cascade of what actually should be internal and ends with controllers.

I was looking this up and ASP.NET doesn't offer this out of the box which is weird because we really shouldn't open up things like configuration, option etc.

Reply:
Overall it's good to have discipline in making things `internal` when they don't need to be

Reply:
Agreed

Reply:
But I think it's still worth the hassle